### PR TITLE
[Network] `az network vnet subnet`: Add parameter `--default-outbound-access`

### PR DESCRIPTION
**Pull request by ZengTaoxu**
**Related command**

**Description**

Resolve: https://github.com/Azure/azure-cli/issues/26956

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Network

File: src/azure-cli/azure/cli/command_modules/network/aaz/latest/network/vnet/subnet/_create.py [hunk 1]
```diff
@@ -927,9 +933,16 @@ def _build_schema_network_interface_ip_configuration_read(cls, _schema):
             serialized_name="provisioningState",
             flags={"read_only": True},
         )
+        properties.sync_mode = AAZStrType(
+            serialized_name="syncMode",
+        )
```
Comment (necusjz) [inline src/azure-cli/azure/cli/command_modules/network/aaz/latest/network/vnet/subnet/_create.py:938]: Related to other feature requests, better to move it to another PR to make the current PR more clear.

**Review by necusjz — APPROVED** [review]

**Review by evelyn-ys — APPROVED** [review]

**Review by yanzhudd — APPROVED** [review]

**Review by jsntcy — APPROVED** [review]

**Review by kairu-ms — APPROVED** [review]
